packages/esm-appointments-app/src/appointments/scheduled/scheduled-appointments.component.tsx

```diff
@@ -16,101 +23,165 @@ interface ScheduledAppointmentsProps {
+      for (const extension of Object.getOwnPropertyNames(allowedExtensions)) {
+        if (allowedExtensions[extension]) {
+          setCurrentTab(extension);
+          break;
+        }
+      }
     }
-  }, [scheduledAppointmentPanels]);
-
-  // set the default tab whenever the date type changes
-  useEffect(() => {
-    setCurrentTab(getDefaultTab(scheduledAppointmentPanels));
-  }, [dateType]);
-
-  const getDefaultTab = (panels) => {
-    return panels?.find(
-      (panel) =>
-        (isDateInPast && panel.config.showForPastDate) ||
-        (isToday && panel.config.showForToday) ||
-        (isDateInFuture && panel.config.showForFutureDate),
-    )?.config.title;
-  };
+  }, [allowedExtensions, currentTab]);
 
   return (
     <>
-      {dateType === 'today' && (
-        <ContentSwitcher className={styles.switcher} size="sm" onChange={({ name }) => setCurrentTab(name)}>
-          {scheduledAppointmentPanels
-            .filter((panel) => panel.config.showForToday)
-            .map((panel, i) => {
-              return <Switch key={i} name={panel.config.title} text={t(panel.config.title)} />;
-            })}
-        </ContentSwitcher>
-      )}
-
-      {dateType === 'pastDate' && (
-        <ContentSwitcher className={styles.switcher} size="sm" onChange={({ name }) => setCurrentTab(name)}>
-          {scheduledAppointmentPanels
-            .filter((panel) => panel.config.showForPastDate)
-            .map((panel, i) => {
-              return <Switch key={i} name={panel.config.title} text={t(panel.config.title)} />;
-            })}
-        </ContentSwitcher>
-      )}
-
-      {dateType === 'futureDate' && (
-        <ContentSwitcher className={styles.switcher} size="sm" onChange={({ name }) => setCurrentTab(name)}>
-          {scheduledAppointmentPanels
-            .filter((panel) => panel.config.showForFutureDate)
-            .map((panel, i) => {
-              return <Switch key={i} name={panel.config.title} text={t(panel.config.title)} />;
-            })}
-        </ContentSwitcher>
-      )}
+      <ContentSwitcher className={styles.switcher} size="sm" onChange={({ name }) => setCurrentTab(name)}>
+        {scheduledAppointmentPanels.filter(shouldShowPanel).map((panel) => {
+          return <Switch key={`panel-${panel.name}`} name={panel.config.title} text={t(panel.config.title)} />;
+        })}
+      </ContentSwitcher>
 
       <ExtensionSlot name={scheduledAppointmentsPanelsSlot}>
-        {(extension) =>
-          extension.config.title === currentTab && (
-            <div className={styles.container}>
-              <Extension
-                state={{
-                  date,
-                  appointmentServiceType,
-                  status: extension.config.status,
-                  title: extension.config.title,
-                }}
-              />
-            </div>
-          )
-        }
+        {(extension) => {
+          return (
+            <ExtensionWrapper
```

Comment:
To make this work, I had to wrap the extension in a new component, which is just responsible for:

1. Rendering the extension (visibly or invisibly) so that we get the configuration
2. Updating the `allowedExtensions` with a computation of whether the `Switcher` for this extension should be rendered.